app/main.py

```diff
@@ -1,2 +1,116 @@
+from __future__ import annotations
+from typing import Any, Hashable, Iterator
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    hash_: int
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.__size = 0
+        self.__capacity = 8
+        self._hash_table: list[Node | None] = [None] * self.__capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash = hash(key)
+        hash_index = self.__index_to_write(key)
+        if self._hash_table[hash_index] is None:
+            self.__size += 1
+        node = Node(hash_=key_hash, key=key, value=value)
+        self._hash_table[hash_index] = node
+
+        if len(self) > int(self.__capacity * (2 / 3)):
+            self.__resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        if not len(self._hash_table):
+            raise KeyError
+        hash_index = self.__index_to_read(key)
+        return self._hash_table[hash_index].value
```

Comment:
The `__getitem__` method should raise a `KeyError` if the key is not found. Currently, it may raise an `AttributeError` if `self._hash_table[hash_index]` is `None`. Consider adding a check for `None` before accessing `.value`.